### DISCUSSION TITLE
Status Updates

### DISCUSSION THREAD

**Discussion by davefojtik** [davefojtik/RunPod-Fooocus-API#14] [Announcements]
### In this thread I'll try to inform you about the state of this repo between updates and releases

> `UPDATE 24/03/2024:` I am basically done with compatibility update with Fooocus-API **0.3.32**, as well as with some new features, including a new param for preview sending from [#11](https://github.com/davefojtik/RunPod-Fooocus-API/issues/11)
> But there were repo owner changes and major code reworks going on in the Fooocus-API repo, and this release contains some major bugs. So I'll wait a bit more and see if we can fix them in the main repo first.

> `UPDATE 08/04/2024:` Some major bugs affecting performance were not solved till today's **0.3.33** release, so instead of making the repo compatible with the version including them, I will skip the **0.3.32** and try to test **0.3.33**.

**Comment by davefojtik** [comment]
> `UPDATE 26/04/2024:` The Fooocus-API repo had a major change lately: directly including the original Fooocus source code. Within about 2 weeks there were 6 releases and many commits basically every day. I waited for it to calm down, and to have a more stable "LTS" release again. Currently testing **0.4.0.6** and some refractor and minor fixes in our code

**Comment by davefojtik** [comment]
> `UPDATE 02/06/2024:` New Fooocus versions are looking good! We have performance and bug fixes, clip skip and an upcoming important nsfw filter feature. That one is still in the dev branch but Fooocus-API merged it already. Commits are already on version 0.4.1.0. I am currently waiting if there's gonna be a release.

**Comment by davefojtik** [comment]
> `Update 11/07/2024:` Following the recent lllyasviel's (Fooocus author) controversy regarding his newly released software **`Paints-UNDO`**, I am sad to announce I decided to pause updates and contributions on this repository or any associated project until it's resolved. Support availability is not affected.

**Comment by pribeh** [comment]
@davefojtik first, thank you for your great work on this repo. It's been a great help to my team. Is there any update on the status of your involvement with this repo? I'm asking in case you think we should fork this repo to adapt it for the newer version of [Fooocus API](https://github.com/mrhan1993/FooocusAPI) or if you'd like our help to adapt this repo

**Reply by davefojtik** [reply]
I am still trying to resolve issues and give support. But there are several reasons I didn't update for a while:
- The [Fooocus-API](https://github.com/mrhan1993/Fooocus-API) itself may be hitting its end of development. Mrhan started working on his own [refractored version](https://github.com/mrhan1993/FooocusAPI) some time ago, implementing the FastAPI right into the Fooocus, not as a wrapper around it. I don't know if it will be merged into the Fooocus, if both API repos will get updates in the future or if there even will be another version of the current one. The refractored version is quite different so it would completely break everyone's code and payloads. So there's that uncertainty about where to go now.
- Second, there has been quite busy feedback regarding the recent controversial release of lllyasviel' [Paints-UNDO](https://github.com/lllyasviel/Paints-UNDO), as mentioned above. I was always amazed by his genius contributions to the AI world as he created Foocus, ICLight, Forge WebUI etc., all the more it hurt to see this one, and how he (didn't) respond to the backlash. This was the ethical part for me.
- And lastly, the updates can be quite time-consuming, especially the testing part. In the past, if I wasn't sure about the next versions I also waited a bit.

That being said, of course, I highly welcome any contribution to this repo! I was hoping for it since I made it public. If you want 4.1.1, or have a take about which way we should go next, I would be very happy to cooperate, or even to become a minority in the contributions. 😅
